src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] list;
+    private int size;
+
+    public ArrayList() {
+        list = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        resizeIfNeeded();
+        list[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index);
+        resizeIfNeeded();
+        System.arraycopy(list,index,list,index + 1,size - index);
+        list[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0;i < list.size();i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexBounds(index);
+        return list[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndexBounds(index);
+        list[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexBounds(index);
+        T removedElement = list[index];
+        System.arraycopy(list,index + 1,list,index,size - index - 1);
+        size--;
+        return removedElement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int index = elementCheck(element);
+        if (index == -1) {
+            throw new NoSuchElementException("No such element");
+        }
+        T removedElement = list[index];
```

Comment:
```suggestion
        return remove(index);
```

Look, at lines 65-68, this is same code as on lines 53-56. After checking index you could just return remove(index);